contentcuration/contentcuration/frontend/channelList/views/Channel/ChannelItem.vue

```diff
@@ -110,29 +110,32 @@
               :to="channelDetailsLink"
             >
               <KIconButton
+                v-if="detailsIcon"
                 :color="$themeTokens.primary"
-                data-test="details-button"
+                :data-test="detailsIcon.dataTest"
                 class="mr-1"
-                icon="info"
-                :tooltip="$tr('details')"
+                :icon="detailsIcon.icon"
+                :tooltip="detailsIcon.tooltip"
               />
             </KRouterLink>
 
             <KIconButton
-              v-if="!allowEdit && channel.published"
+              v-for="config in filteredIcons"
+              :key="config.key"
               class="mr-1"
-              icon="copy"
-              :tooltip="$tr('copyToken')"
-              data-test="token-button"
-              @click.stop.prevent="tokenDialog = true"
+              :icon="config.icon"
+              :tooltip="config.tooltip"
+              :data-test="config.dataTest"
+              @click.stop.prevent="config.clickHandler"
             />
             <ChannelStar
               v-if="loggedIn"
               :channelId="channelId"
               :bookmark="channel.bookmark"
               class="mr-1"
             />
-            <BaseMenu v-if="showOptions">
+
+            <BaseMenu v-if="iconConfigs.find(config => config.key === 'kebab-menu')?.show">
```

Comment:
Flagging that we can't use the conditional chaining operator in vue templates at the moment - so will need to tweak this - can move to a computed prop if that's easier.

This should fix the test failures.